jax/_src/checkify.py

```diff
@@ -862,10 +867,10 @@ def while_loop_error_check(error, enabled_errors, *in_flat, cond_nconsts,
   to_move = [False] * num_error_vals + [True] * cond_nconsts + [False] * len(carry)
   compat_cond_jaxpr = pe.move_binders_to_front(compat_cond_jaxpr_, to_move)
 
-  new_in_flat = [*c_consts, *b_consts, *err_vals, *carry]
+  new_in_flat = [*c_consts, *c_consts, *b_consts, *err_vals, *carry]
   all_out_vals = lax.while_p.bind(
       *new_in_flat, cond_nconsts=cond_nconsts, cond_jaxpr=compat_cond_jaxpr,
-      body_nconsts=body_nconsts, body_jaxpr=checked_body_jaxpr)
+      body_nconsts=body_nconsts+cond_nconsts, body_jaxpr=checked_body_jaxpr)
```

Comment:
```suggestion
      body_nconsts=cond_nconsts+body_nconsts, body_jaxpr=checked_body_jaxpr)
```

That doesn't change the value of the expression but it reinforces that we're putting the original cond consts first.

Reply:
Done!